Enable RRef tests for other RPCAgent

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27789 Enable RRef tests for other RPCAgent**

Differential Revision: [D5444828](https://our.internmc.facebook.com/intern/diff/D5444828/)

xush6528 committed:
Enable RRef tests for other RPCAgent



Differential Revision: [D5444828](https://our.internmc.facebook.com/intern/diff/D5444828/)

[ghstack-poisoned]

rohan-varma commented on test/rpc_test.py:
```diff
@@ -660,104 +660,96 @@ def _test_multi_remote_call(self, fn, args_fn=lambda x: (), kwargs_fn=lambda x:
             self.assertEqual(rrefs[i].to_here(), expected[i])
 
     @dist_init
-    @requires_process_group_agent
```
If we're removing this decorator, should we also remove the function definition for it?

xush6528 replied:
@rohan-varma Updated the PR. I refurbished this decorator to make it useful.

rohan-varma commented:
I'm not sure if we should completely eliminate the decorator, since @mrshenli 's intent seemed to be to not run the test if the agent is an `RpcAgent` but not specifically a `ProcessGroupAgent`. If we still need to keep this functionality, we could possibly pass this information directly to the `dist_init` decorator.

mrshenli commented:
> since @mrshenli 's intent seemed to be to not run the test if the agent is an RpcAgent but not specifically a ProcessGroupAgent.

Yes, I was trying to avoid running these tests with other types of RpcAgent, as they don't have proper termination detection. If we land this now, will it break internal tests? If yes, shall we first fix the order of the wrappers to starting running those tests and come back later to remove it when we are certain that other types of RpcAgent would terminate properly.

BTW, do you know why `unittest.skipUnless` shows pass instead of skip on skipped tests with pytest?

xush6528 committed:
Update on "Enable RRef tests for other RPCAgent"


Differential Revision: [D5444828](https://our.internmc.facebook.com/intern/diff/D5444828/)

[ghstack-poisoned]

mrshenli commented:
`short_perf_test_gpu` is disabled on master. This should be good to go after fixing lint

mrshenli commented on test/rpc_test.py:
```diff
@@ -16,13 +16,13 @@
+def requires_process_group_agent(message=""):
+    def decorator(old_func):
+        return unittest.skipUnless(
+            TEST_CONFIG.backend == RpcBackend.PROCESS_GROUP,
```
This is still `TEST_CONFIG.rpc_backend` on master, where did we change that to `TEST_CONFIG.backend`?

xush6528 replied:
Rebasing

mrshenli commented on test/rpc_test.py:
```diff
@@ -16,13 +16,13 @@
-        isinstance(_agent, ProcessGroupAgent),
-        "Only ProcessGroupAgent supports global termination detection",
-    )
+def requires_process_group_agent(message=""):
```
Does this decorator run tests for you? I found it still skips tests for me, and the following works:

```
def requires_process_group_agent(func):
    from torch.distributed.rpc.api import _agent

    return unittest.skipUnless(
        TEST_CONFIG.rpc_backend == RpcBackend.PROCESS_GROUP,
        "Only ProcessGroupAgent supports global termination detection",
    )(func)
```

xush6528 replied:
@mrshenli 

It's working for me.

xush6528 committed:
Update on "Enable RRef tests for other RPCAgent"


Differential Revision: [D5444828](https://our.internmc.facebook.com/intern/diff/D5444828/)

[ghstack-poisoned]
